sdlib/d/gc/hpd.d

```diff
@@ -111,6 +111,46 @@ public:
 		return bestIndex;
 	}
 
+	bool reserve(uint index, uint pages) {
+		assert(pages > 0 && pages <= PageCount, "Invalid number of pages!");
+		assert(index <= PageCount - pages, "Invalid index!");
+
+		if (pages > longestFreeRange) {
+			return false;
+		}
+
+		uint freePos, freeLength;
+		allocatedPages.nextFreeRange(index, freePos, freeLength);
+
+		if ((freePos != index) || (freeLength < pages)) {
+			return false;
+		}
+
+		allocatedPages.setRange(index, pages);
+		usedCount += pages;
+
+		if (freeLength != longestFreeRange) {
+			return true;
+		}
+
+		// If allocated from the longest free range, compute the new one:
+		auto longestFree = longestFreeRange - pages;
+		uint i = 0;
+		// FIXME: could stop at PageCount - longestFree
+		while (i < PageCount
```

Comment:
Alternatively, you can do `i < PageCount - longestFree`, but this should come with appropriate testing. Or a comment explaining this could be done.